lua/kulala/parser/curl.lua

```diff
@@ -0,0 +1,78 @@
+local Shlex = require("kulala.shlex")
+
+local M = {}
+
+---Parse a curl command into a Request object
+---@param curl string The curl command line to parse
+---@return Request|nil -- Table with a parsed data or nil if parsing fails
```

Comment:
I get an error for underfined Request. Can I use the one from `init.lua` or better define own type here?

Reply:
I think you can use the one from `init.lua`.

Reply:
I am still new to Lua - I wanted to share annotations, which sounds like a complicated topic in Lua - https://github.com/LuaLS/lua-language-server/wiki/Annotations. Maybe I'll drop the annotations from the code. They're not important for feature to work.